src/python/WMComponent/DBS3Buffer/DBSBufferBlock.py

```diff
@@ -141,7 +141,11 @@ def addFile(self, dbsFile, datasetType, primaryDatasetType):
         lumiList = []
         for run in dbsFile.getRuns():
             for lumi in run.lumis:
-                lumiList.append({'lumi_section_num': lumi, 'run_num': run.run})
+                dbsLumiDict = {'lumi_section_num': lumi, 'run_num': run.run}
+                if run.getEventsByLumi(lumi) is not None:
```

Comment:
After we updates the patch there will be still many cases event_count is None (old cmssw release). Yuyi said DBS make extra call if event_count exists (whether it is None or not). In case there are a lot of lumis this might cause performance problem.